runtime/compiler/aarch64/codegen/J9TreeEvaluator.cpp

```diff
@@ -39,3 +51,43 @@ J9::ARM64::TreeEvaluator::generateFillInDataBlockSequenceForUnresolvedField(TR::
    TR_ASSERT_FATAL(false, "This helper implements platform specific code for Fieldwatch, which is currently not supported on ARM64 platforms.\n");
    }
 	
+TR::Register *J9::ARM64::TreeEvaluator::DIVCHKEvaluator(TR::Node *node, TR::CodeGenerator *cg)
+   {
+   TR::Node *divisor = node->getFirstChild()->getSecondChild();
+   TR::Register *srcReg = NULL;
+   TR::Register *zeroReg = NULL;
+   bool is64Bit = node->getType().isInt64();
+   bool isConstDivisor = divisor->getOpCode().isLoadConst();
+   TR::LabelSymbol *snippetLabel = NULL;
+
+   if (!isConstDivisor || (!is64Bit && divisor->getInt() == 0) || (is64Bit && divisor->getLongInt() == 0))
+      {
+      snippetLabel = generateLabelSymbol(cg);
+      cg->addSnippet(new (cg->trHeapMemory()) TR::ARM64HelperCallSnippet(cg, node, snippetLabel, node->getSymbolReference()));
+
+      if (isConstDivisor)
+         {
+         // No explicit check required
+         generateLabelInstruction(cg, TR::InstOpCode::b, node, snippetLabel, (TR::Instruction*)NULL);
```

Comment:
Don't pass in `NULL` explicitly for the optional parameter as it will default to that anyway.  It just looks a little cleaner without it.

Reply:
I think I originally did this because of a compile error that came up about the function call being ambiguous. I'll take a look at it again when I do my next build, though!

Reply:
Hm.  That shouldn't be.  There is precedent in other places for successfully writing it this way.  e.g.,
https://github.com/eclipse/omr/blob/f284dd273f3226c013a3e19c1d5c0ea331447420/compiler/aarch64/codegen/ControlFlowEvaluator.cpp#L85

Reply:
Yes, I thought it was very strange when it happened. I will see soon if I can replicate it, and let you know.